keras/src/models/model.py

```diff
@@ -556,6 +556,132 @@ def _get_variable_map(self):
         map_saveable_variables(self, store=store, visited_saveables=set())
         return store
 
+    def get_state_tree(
+        self,
+        trainable_variables=False,
+        non_trainable_variables=False,
+        optimizer_variables=False,
+        metrics_variables=False,
+    ):
+        """Retrieves tree-like structure of specified variables of the model.
+
+        This method allows selective retrieval of different variables
+        (trainable, non-trainable, optimizer, and metrics) associated with the
+        model. The variables are returned in a nested dictionary format,
+        where the keys correspond to the variable names and the values are the
+        nested representations of the variables.
+
+        Args:
+            trainable_variables: Whether to include trainable variables.
+            non_trainable_variables: Whether to include non-trainable variables.
+            optimizer_variables: Whether to include optimizer variables.
+            metrics_variables: Whether to include metrics variables.
+
+        Returns:
+            dict: A dictionary containing the nested representations of the
+                requested variables. The keys are the variable names, and the
+                values are the corresponding nested dictionaries. If no variable
+                types are requested, an empty dictionary is returned.
+        """
+        variables = {}
+        if trainable_variables:
+            variables["trainable_variables"] = self._create_nested_dict(
+                self.trainable_variables
+            )
+        if non_trainable_variables:
+            variables["non_trainable_variables"] = self._create_nested_dict(
+                self.non_trainable_variables
+            )
+        if optimizer_variables:
+            variables["optimizer_variables"] = self._create_nested_dict(
+                self.optimizer.variables
+            )
+        if metrics_variables:
+            variables["metrics_variables"] = self._create_nested_dict(
+                self.metrics_variables
+            )
+
+        return variables
+
+    def _create_nested_dict(self, variables):
+        flat_dict = {}
+        for v in variables:
+            if v.path in flat_dict:
+                raise ValueError(
+                    "The following variable path is found twice in the model: "
+                    f"'{v.path}'. The state tree can only be constructed when "
```

Comment:
Rather then "the state tree," reference the actual exposed function `get_state_tree()` can only be called"

Reply:
Done